### DISCUSSION TITLE
Welcome to mergerfs-tools Discussions!

### DISCUSSION THREAD

**Discussion by trapexit** [trapexit/mergerfs-tools#127] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by schizovivek** [comment]
Unsure if this is the right place but I had a query regarding the mergerfs.balance tool.
I had a pool of 4 disks to which I just added a fifth. I have a windows machine with Drivepool and assumed that mergerfs rebalanced everything by default. After a couple days of the new drive staying relatively empty I searched on how to rebalance and found this repo.

Last night I started the the balancing tool and noticed that it started to move data from other disks into the new one. Since all drives are 4 tb and the 4 drives were close to being full this was taking a long time. I had stopped most the services that were accessing the pool but looks like some were still on. Just a couple hours ago the sync failed due to permission being denied on some file (note: I ran the command without sudo which is what most likely caused the issue)

My question here is: My pool if I recall correctly had between 3.5 to 3.8 TB space remaining. Now that the balance stopped I can see only 2.4 TB remaining. Could it be that there are left over files that didn't get deleted? I have no idea how rsync works but does the rsync delete files at end of the activity?

The command I ran:
`./mergerfs.balance /mnt/pool`

Output of the info command: `./mergerfs.ctl info`
```
- mount: /mnt/pool
  version: 2.33.5
  pid: 683
  srcmounts:
    - /mnt/disk1
    - /mnt/disk2
    - /mnt/disk3
    - /mnt/disk4
    - /mnt/disk5
 ```

These are the last few lines from the balance failure log
```
file: personal/immich_library_test/encoded-video/8c517971-87a1-473b-b63b-2d85dc72151e/be/4a/be4a9d6c-af94-4868-8933-d54373e3c7e6.mp4
from: /mnt/disk5
to:   /mnt/disk1
rsync -avlHAXWE --relative --progress --remove-source-files /mnt/disk5/./personal/immich_library_test/encoded-video/8c517971-87a1-473b-b63b-2d85dc72151e/be/4a/be4a9d6c-af94-4868-8933-d54373e3c7e6.mp4 /mnt/disk1/
sending incremental file list
rsync: [generator] failed to set times on "/mnt/disk1/personal/immich_library_test/encoded-video": Operation not permitted (1)
rsync: [generator] recv_generator: mkdir "/mnt/disk1/personal/immich_library_test/encoded-video/8c517971-87a1-473b-b63b-2d85dc72151e" failed: Permission denied (13)
*** Skipping any contents from this failed directory ***
personal/
personal/immich_library_test/
personal/immich_library_test/encoded-video/
personal/immich_library_test/encoded-video/8c517971-87a1-473b-b63b-2d85dc72151e/

sent 452 bytes  received 30 bytes  964.00 bytes/sec
total size is 425,676,734  speedup is 883,146.75
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1338) [sender=3.2.7]
ERROR - exited with exit code: 23
Branches within 2.0% range:
 * /mnt/disk4: 12.57% free
 * /mnt/disk3: 12.57% free
 * /mnt/disk2: 12.63% free
 * /mnt/disk1: 16.24% free
```

Does the dedup command look for normal duplicate files or duplicate files caused by mergerfs specifically? Would using that utility be the right way to go?

**Reply by trapexit** [reply]
> Unsure if this is the right place

It's a forum like setup. Just create a new thread. No reason to put questions on the welcome post.

> I have no idea how rsync works but does the rsync delete files at end of the activity?

It doesn't remove the previous till it knows it was copied correctly. And naturally that is based on permissions. If you didn't run it as root then it won't have proper perms.

> Does the dedup command look for normal duplicate files or duplicate files caused by mergerfs specifically? Would using that utility be the right way to go?

It deletes files that overlap the namespace. No, it does not work like a normal dedup app because there would be no point. There are dozens of those. It works on mergerfs specific facts. Like that two branches can have the same relative path. And sure... that is one way to clean up.

But why are you rebalancing? That really shouldn't be necessary. The point of the app is to help people who inadvertently chose epmfs and wanted mfs.

**Comment by schizovivek** [comment]
> No reason to put questions on the welcome post.

Apologies for that. I missed the last line and thought this was a thread to be used for discussions.

>  It works on mergerfs specific facts

This is exactly what I was hoping it did but just wanted to confirm before running the command. It's only after I posted did I realize that the -e param is needed to actually delete so I was able to test out what it'd have done and found about 25 files that it cleared up (not the cause of the spike)

I've been looking at the data sets across the pool and don't see anything wrong as of now. Looks like something else could have caused the sudden bloat. Looking at Syncthing and the downloaders. I'll create a separate thread if I find something and feel it's related to mergerfs.

Thanks for your response.
